### PR TITLE
4.6 encryption/decryption methods

### PR DESCRIPTION
**Pull request by jeff-fichtner**
@bontgoy thanks for the help!
@kimberlysuazo

### CODE REVIEW

File: ruby/secret_agents.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	-return the full return string
+-end method
+=end
+
+#The encrypt method returns the next letter in the alphabet for each inputed value in string.
+def encrypt str
+  index = 0
+  return_str = ""
+  until index >= str.length
+    if str[index] != " " && str[index] != "z"
```
Comment (Bontgoy) [inline ruby/secret_agents.rb:26]: Think order of operations here and how many times we want our code to check an edge case.  

```
x = str[index]

if x != " "
     if x == "z"
          return_str += "a"
     else
          return_str += x.next
     end
else
     return_str += x
end
```

Comment (Bontgoy) [inline ruby/secret_agents.rb:26]: think DRY(do not repeat yourself) when coming up with conditions and handling edge case.

Comment (jeff-fichtner) [inline ruby/secret_agents.rb:26]: @bontgoy so what you're saying is simplifying the control flow portion so as to not repeat my conditions as much. in terms of order of operations, what do you mean exactly?

Comment (Bontgoy) [inline ruby/secret_agents.rb:26]: Speaking in terms of the name question.

If the names are vampire then we know they are vampires.  No need to run anymore code.  But optimizing your control flow you can allow your compiler from running a lot of extra code

**Pull request merged**